src/components/HomeTab/index.jsx

```diff
@@ -88,33 +89,53 @@ HomeTab.propTypes = {
       internal_name: PropTypes.string,
     }),
   ).isRequired,
-  investGetSpec: PropTypes.func.isRequired,
-  loadState: PropTypes.func.isRequired,
-  recentSessions: PropTypes.array,
+  openInvestModel: PropTypes.func.isRequired,
+  recentJobs: PropTypes.arrayOf(
+    PropTypes.array
+  ),
 };
 HomeTab.defaultProps = {
-  recentSessions: [],
+  recentJobs: [],
 };
 
 /**
  * Renders a button for each recent invest job.
  */
 class RecentInvestJobs extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(jobDataPath) {
+    const jobData = JSON.parse(
+      fs.readFileSync(jobDataPath, 'utf8')
+    );
+    this.props.openInvestModel(
+      jobData.modelRunName,
+      jobData.argsValues,
+      jobData.logfile,
+      jobData.status,
+    );
+  }
+
   render() {
     // Buttons to load each recently saved state
     const recentButtons = [];
-    const { recentSessions } = this.props;
-    recentSessions.forEach((session) => {
+    const { recentJobs } = this.props;
+    recentJobs.forEach((job) => {
       // These properties are required, if they don't exist,
-      // the session data was corrupted and should be skipped
-      let name;
+      // the job's data was corrupted and should be skipped
+      let jobID;
       let metadata;
       let model;
       let workspaceDir;
+      let jobDataPath;
       try {
-        [name, metadata] = session;
+        [jobID, metadata] = job;
         model = metadata.model;
         workspaceDir = metadata.workspace.directory;
+        jobDataPath = metadata.jobDataPath;
       } catch (error) {
         return;

```

Comment:
Good question. The `try/catch` is here because the  `job` and `metadata` objects were parsed from a JSON file that is storing data about a recent /saved session. So this is kind of guarding against a poorly formed JSON that is missing some of those properties. In that case we can catch the `TypeError` and omit that job from the list of recent job buttons. 

I cleaned this up a little bit, but also expect it could change again because we might want to formalize that JSON file into an actual database with a well-defined schema. All the persistent storage stuff is still fairly primitive.